src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -86,5 +113,260 @@ private protected IReadOnlyList<PackageReference> GetTransitivePackageReferences
+                        userInstalled: false,
+                        developmentDependency: false,
+                        requireReinstallation: false,
+                        allowedVersions: dep.VersionRange);
+
+                    if (!visited.Contains(pkgChild.PackageIdentity))
+                    {
+                        MarkTransitiveOrigin(top, pkgChild, graph, visited, fxRidEntry, token);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Generates a cache key for Transitive Originas cache
+        /// </summary>
+        /// <param name="transitivePackage"></param>
+        /// <returns>A string with given key</returns>
+        /// <seealso cref="GetCachedTransitiveOrigin(PackageIdentity)"/>
+        /// <seealso cref="SetCachedTransitiveOrigin(PackageIdentity, TransitiveEntry)"/>
+        internal PackageIdentity GetTransitiveCacheKey(PackageIdentity transitivePackage)
+        {
+            return transitivePackage;
+        }
+
+        /// <summary>
+        /// Obtains cached entry for a given transitive package
+        /// </summary>
+        /// <param name="transitivePackage">Identity of transitive package</param>
+        /// <returns>A <see cref="TransitiveEntry"/> object, or <c>null</c> if not found</returns>
+        /// <seealso cref="ClearCachedTransitiveOrigins"/>
+        /// <seealso cref="SetCachedTransitiveOrigin(PackageIdentity, TransitiveEntry)"/>
+        internal TransitiveEntry GetCachedTransitiveOrigin(PackageIdentity transitivePackage)
+        {
+            var key = GetTransitiveCacheKey(transitivePackage);
+
+            if (TransitiveOriginsCache.ContainsKey(key))
+            {
+                return TransitiveOriginsCache[key];
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Replaces cached entry for a given transitive package with a <see cref="TransitiveEntry"/>
+        /// </summary>
+        /// <param name="transitivePackage">Identity of transitive package</param>
+        /// <param name="origins">Packages identified as package origins</param>
+        /// <seealso cref="ClearCachedTransitiveOrigins"/>
+        /// <seealso cref="GetCachedTransitiveOrigin(PackageIdentity)"/>
+        internal void SetCachedTransitiveOrigin(PackageIdentity transitivePackage, TransitiveEntry origins)
+        {
+            var key = GetTransitiveCacheKey(transitivePackage);
+            TransitiveOriginsCache[key] = origins;
+        }
+
+        /// <summary>
+        /// Clears Transitive Origins cache
+        /// </summary>
+        /// <seealso cref="GetCachedTransitiveOrigin(PackageIdentity)"/>
+        /// <seealso cref="SetCachedTransitiveOrigin(PackageIdentity, TransitiveEntry)"/>
+        internal void ClearCachedTransitiveOrigins()
+        {
+            TransitiveOriginsCache.Clear();
+            IsInstalledAndTransitiveComputationNeeded = true;
```

Comment:
This call is duplicated. 

You'd only call this method if `IsInstalledAndTransitiveComputationNeeded` is true to begin with. 

Removing this call woudl allow you to remove this method.

Reply:
Removed method.